packages/features/schedules/lib/use-schedule/useSchedule.ts

```diff
@@ -29,24 +29,31 @@ export const useSchedule = ({
   return trpc.viewer.public.slots.getSchedule.useQuery(
     {
       usernameList: getUsernameList(username ?? ""),
-      eventTypeSlug: eventSlug!,
+      // Prioritize slug over id, since slug is the first value we get available.
+      // If we have a slug, we don't need to fetch the id.
+      // TODO: are queries using eventTypeId faster? Even tho we lost time fetching the id with the slug.
+      ...(eventSlug ? { eventTypeSlug: eventSlug } : { eventTypeId: eventId ?? 0 }),
       // @TODO: Old code fetched 2 days ago if we were fetching the current month.
       // Do we want / need to keep that behavior?
       startTime: monthDayjs.startOf("month").toISOString(),
       // if `prefetchNextMonth` is true, two months are fetched at once.
       endTime: (prefetchNextMonth ? nextMonthDayjs : monthDayjs).endOf("month").toISOString(),
       timeZone: timezone!,
-      eventTypeId: eventId!,
       duration: duration ? `${duration}` : undefined,
     },
     {
+      trpc: {
+        context: {
+          skipBatch: true,
+        },
+      },
       refetchOnWindowFocus: false,
       enabled:
         Boolean(username) &&
-        Boolean(eventSlug) &&
-        (Boolean(eventId) || eventId === 0) &&
```

Comment:
i don't think you should remove eventId from here because it is required on the backend https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/slots/util.ts#L189. only dynamic event types have eventId.

Reply:
Didn't removed just moved to last conditional